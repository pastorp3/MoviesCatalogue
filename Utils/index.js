import { checkPropTypes } from 'prop-types';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import RootReducer from '../src/reducers/index';

export const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return component;
};

export const checkProps = (component, expectedProps) => {
  const propsErr = checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
  return propsErr;
}
