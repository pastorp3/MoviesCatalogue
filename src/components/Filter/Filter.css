.filter-container {
  padding-top: 30px;
  text-align: right;
  padding-right: 150px;
}

.filter-text {
  margin-right: 10px;
  font-family: sans-serif;
}
