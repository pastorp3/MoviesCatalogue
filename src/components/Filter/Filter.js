/* eslint-disable react/forbid-prop-types, react/jsx-closing-tag-location, react/jsx-wrap-multilines, max-len */
import PropTypes from 'prop-types';
import './Filter.css';

const Filter = ({ data, change }) => (
  (
    <div className="filter-container">
      <span className="filter-text">Filter by genre</span>
      {
        data && <select name="genre" onChange={(event) => change(event)}>
          <option value="All">All</option>
          {data.map((genre) => <option value={genre.id} key={genre.id}>{genre.name}</option>)}
          </select>
      }
    </div>
  )
);

Filter.propTypes = {
  data: PropTypes.array.isRequired,
  change: PropTypes.func.isRequired,
};

export default Filter;

/* eslint-enable */
