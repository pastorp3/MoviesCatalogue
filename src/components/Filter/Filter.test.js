/* eslint-disable  preact/jsx-props-no-spreading */
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Filter from './Filter';
import { checkProps, findByTestAtrr } from '../../../Utils';

const testFunc = (e) => e;

const setUp = (props = {
  data: [],
  change: testFunc,
}) => {
  const component = shallow(<Filter {...props} />);
  return component;
};

describe('Filter Component', () => {
  describe('Checking ProrTypes', () => {
    it('Should not throw a warning', () => {
      const expectedProps = {
        data: [],
        change: testFunc,
      };
      const propsErr = checkProps(Filter, expectedProps);
      expect(propsErr).toBe(undefined);
    });
  });

  let component;
  beforeEach(() => {
    component = setUp();
  });

  it('Should render whitout error', () => {
    const wrappre = findByTestAtrr(component, 'Filter');
    expect(wrappre.length).toBe(1);
  });

  it('Should render select element ', () => {
    const wrapper = findByTestAtrr(component, 'Filter-select');
    expect(wrapper.length).toBe(1);
  });

  it('Should render according with snapshout whithout error', () => {
    const props = {
      data: [],
      change: testFunc,
    };
    const tree = renderer.create(<Filter {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

/* eslint-enable */
