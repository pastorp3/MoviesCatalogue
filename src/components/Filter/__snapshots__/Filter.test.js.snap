// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component Should render according with snapshout whithout error 1`] = `
<div
  className="filter-container"
  data-test="Filter"
>
  <span
    className="filter-text"
  >
    Filter by genre
  </span>
  <select
    data-test="Filter-select"
    name="genre"
    onChange={[Function]}
  >
    <option
      value="All"
    >
      All
    </option>
  </select>
</div>
`;
